src/eventbridge/eventpattern/index.ts

```diff
@@ -0,0 +1,798 @@
+        }
+        // If one of the patterns are exists: true, return exists: true
+        if (pattern1.isPresent) {
+          return pattern1;
+        }
+      }
+      if (isPresentPattern(pattern2)) {
+        // If one of the patterns are exists: true, return exists: true
+        if (pattern2.isPresent) {
+          return pattern1;
+        }
+      }
+      if (isNumericAggregationPattern(pattern1)) {
+        if (isNumericAggregationPattern(pattern2)) {
+          return reduceNumericAggregate({
+            ranges: [...pattern1.ranges, ...pattern2.ranges],
+          });
+        }
+        if (isNumericRangePattern(pattern2)) {
+          return reduceNumericAggregate({
+            ranges: [...pattern1.ranges, pattern2],
+          });
+        }
+        throw Error("Cannot OR a numeric ranges with any other pattern.");
+      }
+      if (isNumericAggregationPattern(pattern2)) {
+        if (isNumericRangePattern(pattern1)) {
+          return reduceNumericAggregate({
+            ranges: [...pattern2.ranges, pattern1],
+          });
+        }
+        throw Error("Cannot OR a numeric ranges with any other pattern.");
+      }
+      if (isNumericRangePattern(pattern1)) {
+        if (isNumericRangePattern(pattern2)) {
+          return mergeNumericOr(pattern1, pattern2);
+        }
+        throw Error("Cannot OR a numeric ranges with any other pattern.");
+      }
+      // TODO support OR logic with exists: false
+      if (isNumericRangePattern(pattern2)) {
+        throw Error("Cannot OR a numeric ranges with any other pattern.");
+      }
+      return {
+        patterns: [
+          ...(isAggregatePattern(pattern1)
+            ? pattern1.patterns
+            : isEmptyPattern(pattern1)
+            ? []
+            : [pattern1]),
+          ...(isAggregatePattern(pattern2)
+            ? pattern2.patterns
+            : isEmptyPattern(pattern2)
+            ? []
+            : [pattern2]),
+        ],
+      };
+    }
+    // we don't know how to do this (yet?)
+    throw new Error(`Patterns of key ${key} defined at different levels.`);
+  };
+
+  const handleNumericRange = (expr: BinaryExpr): PatternDocument => {
+    const { eventReference, eventExpr, other, op } = assertOneEventReference(
+      expr.left,
+      expr.right,
+      expr.op
+    );
+
+    if (eventExpr.type !== "number") {
+      throw new Error(
+        "Numeric range only supported for event properties of type number"
+      );
+    }
+
+    const value = assertNumber(getConstant(other)?.value);
+
+    assertValidEventRefererence(eventReference);
+    const range = createSingleNumericRange(value, op);
+    if (range) {
+      return eventReferenceToPatternDocument(eventReference, range);
+    }
+
+    throw Error(`Unsupported numeric range operation: ${op}.`);
+  };
+
+  /**
+   * Any call that we encouter.
+   * Event bridge pattern match calls must all be results at build time.
+   *
+   * Limited to just Includes and StartsWith.
```

Comment:
This will need to be clearly documented in the `README.md`. Anywhere we can document in the code so it will end up in `tsdoc` generated documentation?

Reply:
It would need to be at the `EventBusRule` level.